static/src/javascripts-legacy/projects/commercial/modules/hosted/onward-journey-carousel.js

```diff
@@ -64,8 +68,12 @@ define([
 
     };
 
-    function init() {
-        new HostedCarousel().bindButtons();
+    function init(moduleName) {
+        if (qwery('.js-carousel-pages').length) {
+            performanceLogging.moduleStart(moduleName);
```

Comment:
Yes, I think I made a mistake by moving everything inside the modules. There still is two different behaviours and they should be processed differently. One is:

```
start();
module().then(stop);
```

The other is

```
module(start, stop);
```

I will try to refactor by taking this into account, I think the code will be more readable that way. Perhaps this is a good user case to try [railway-oriented programming](https://fsharpforfunandprofit.com/rop/)